src/convertToObject.js

```diff
@@ -6,7 +6,48 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const arrSoursString = sourceString;
+  const result = {};
+
+  for (let i = 0; i < arrSoursString.length; i++) {
+    if (arrSoursString[i] === ':') {
+      result[`${suchKey(arrSoursString, i)}`] = such(arrSoursString, i);
```

Comment:
The function `suchKey` is not defined in the provided code. Ensure that you have a function that extracts the key from the string correctly.